### PR TITLE
Add a Web deployment step to the tutorial

### PR DESCRIPTION
**Pull request by freakboy3742**
Adds a web deployment step to the tutorial.

Requires a slight reorganisation of steps, as we need to demonstrate web deployment before we introduce the `httpx` dependency into the app.

## PR Checklist:

- [x] All new features have been tested
- [x] All new features have been documented
- [x] I have read the **CONTRIBUTING.md** file
- [x] I will abide by the code of conduct

### CODE REVIEW

**Comment by freakboy3742** [comment]
Two problems preventing this from being mergable:

* Toga-web 0.3.0dev39 doesn't list toga-core as a dependency
* There's an intermittent race condition bug with PyScript not picking up the metadata for packages, leading to an error message about the app not having an app name.

File: docs/tutorial/tutorial-8.rst [hunk 1]
```diff
@@ -1,18 +1,150 @@
...
 
 Next steps
 ==========
 
-We've now got an application with a custom icon on our desktop and phone! Is
-there anywhere else we can deploy a BeeWare app? Turn to :doc:`Tutorial 9
-<tutorial-9>` to find out...
+We've now got an application that is slick and responsive, even when it's
+waiting on a slow API. But it still looks like a tutorial app. Is there anything
+we can do about that? Turn to :doc:`Tutorial 9 <tutorial-9>` to find out...
```
Comment (mhsmith) [inline docs/tutorial/tutorial-8.rst:150]: If pages 9 and 10 aren't ready yet, I think it would be better not to raise hopes and then disappoint them.

**Review by mhsmith — COMMENTED** [review]
> Requires a slight reorganisation of steps, as we need to demonstrate web deployment before we introduce the `httpx` dependency into the app.

Is that because HTTP access from PyScript would be blocked by the same origin policy?

File: docs/tutorial/tutorial-7.rst [hunk 2]
```diff
@@ -106,45 +57,408 @@ it looks like this::
...
+In our case, we want ``httpx`` to be installed on all platforms, so we use the
+app-level ``requires`` setting. The app-level dependencies will always be
+installed; the platform-specific dependecies are installed *in addition* to the
+app-level ones.
+
+.. admonition:: Python only on mobile (for now!)
+
+    On desktop platforms (macOS, Windows, Linux), any ``pip``-installable can
+    be added to your requirements. On mobile platforms, your options are a
+    little more limited - you can only use *pure Python* packages i.e.,
```
Comment (mhsmith) [inline docs/tutorial/tutorial-7.rst:378]: This section should be replaced with a link to [the Briefcase FAQ](https://briefcase.readthedocs.io/en/latest/background/faq.html#can-i-use-third-party-python-packages-in-my-app).

File: docs/tutorial/tutorial-8.rst (hunk 1 above)
Comment (freakboy3742) [reply docs/tutorial/tutorial-8.rst:150]: My original thinking was that we could/should tease what was possible; however, you're probably right that it's better to stop the trail here until we actually fill out those tutorial pages (and, we should actually fill out those tutorial pages!)

**Comment by freakboy3742** [comment]
> Is that because HTTP access from PyScript would be blocked by the same origin policy?

It's actually more fundamental than that - the issue is that there isn't a `socket` library in Pyodide (see the note on [this FAQ entry](https://pyodide.org/en/stable/usage/faq.html#how-can-i-load-external-files-in-pyodide) and [this roadmap item](https://pyodide.org/en/stable/project/roadmap.html#write-http-client-in-terms-of-web-apis)). As a result the Python `http.client` library isn't available, and thus neither is `urllib3`, `requests`, `httpx`, etc.

**Comment by freakboy3742** [comment]
Closing this PR in favour of #176; the PR needs to be on the home repo so I can chain the testing tutorial on top of this one.

**Pull request closed without merging**